catalog/dags/providers/provider_api_scripts/auckland_museum.py

```diff
@@ -0,0 +1,143 @@
+"""
+Content Provider:       AucklandMuseum
+
+ETL Process:            Use the API to identify all CC licensed media.
+
+Output:                 TSV file containing the media and the
+                        respective meta-data.
+
+Notes:                  https://api.aucklandmuseum.com/
+
+Resource:               https://api.aucklandmuseum.com/
+                        https://github.com/AucklandMuseum/API/wiki/Tutorial
+"""
+import logging
+
+from common.constants import IMAGE
+from common.licenses import get_license_info
+from common.loader import provider_details as prov
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
+
+
+logger = logging.getLogger(__name__)
+
+LANDING_URL = (
+    "https://www.aucklandmuseum.com/collections-research/collections/record/am_"
+)
+
+
+class AucklandMuseumDataIngester(ProviderDataIngester):
+    providers = {
+        "image": prov.AUCKLAND_MUSEUM_IMAGE_PROVIDER,
+    }
+    endpoint = "https://api.aucklandmuseum.com/search/collectionsonline/_search"
+    license_url = "https://creativecommons.org/licenses/by/4.0/"
+    delay = 4
+    from_start = 0
+    total_amount_of_data = 10000
+    DEFAULT_LICENSE_INFO = get_license_info(license_url=license_url)
+
+    def get_next_query_params(self, prev_query_params: dict | None, **kwargs) -> dict:
+        # On the first request, `prev_query_params` will be `None`. We can detect this
+        # and return our default params.
+        if not prev_query_params:
+            # Return default query params on the first request
+            # primaryRepresentation contain a image url for each data
+            # "+" is a query string syntax for must be present
+            # copyright:CC state Creative Commons Attribution 4.0
+            return {
+                "q": "_exists_:primaryRepresentation+copyright:CC",
+                "size": "100",
```

Comment:
The max amount of data retrieved from the api is `10,000`. Look at `hits -> total -> value`([api](https://api.aucklandmuseum.com/search/collectionsonline/_search?q=_exists_:primaryRepresentation+copyright:CC&size=1&from=200)) `Size` is the amount of data to present in a get request. `From` is the index of the total value. We can keep incrementing `From` till it reach `10,000` and `get_should_continue` function should know when it reach the limit.

Reply:
I meant that by default `self.batch_limit` is 100, and that you can just say:

```
    "size": self.batch_limit,
```

Rather than hard-coding it separately here.

Reply:
I will update "size" with batch_limit.